ggshield/scan/scan_context.py

```diff
@@ -15,6 +17,9 @@ class ScanContext:
 
     def __post_init__(self) -> None:
         self.command_id = str(uuid.uuid4())
+        self.os = platform.system()
+        self.os_version = platform.version()
```

Comment:
`platform.version()` is not really useful on Linux. You can get more useful information for the `os_name` and `os_version` fields by parsing `/etc/os-release` and reading the ID and VERSION_ID fields.

I suggest you add a helper function at the top of the file to do something along those lines:

```python
def get_os_info() -> (str, str):
    if linux:
        return parse_os_release()
    else:
        # Use lower() to match the linux output
        return platform.system().lower(), platform.version()

def parse_os_release() -> (str, str):
    # ...
```

Then use it like this: `self.os_name, self.os_version = get_os_info()`

Reply:
Thank you for this review @agateau-gg 

I did not know about the `/etc/os-release` trick before, thanks :pray: 

Let me know what you think about the new changes :smile_cat: